src/components/Switch/Switch.js

```diff
@@ -1,35 +1,54 @@
-import React, { Component } from 'react';
+import React, { Component, useState } from 'react';
 
 import './Switch.css';
 
-export default class Switch extends Component {
-  state = {
-    checked: false,
-  };
+function SwitchLabel({labelID, label}){
+  return (
+    <>
+      <label id={labelID} htmlFor={labelID}>{label}</label>
+    </>
+  );
+};
+
+function SwitchInput({onToggle, isEnabled, switchID, switchType, switchTheme}){
+  const [isChecked, setIsChecked] = useState(isEnabled);
 
-  toggleSwitch = () => {
-    this.setState({
-      checked: !this.state.checked,
-    });
-    this.props.onToggle(!this.state.checked);
+  const toggleSwitch = () => {
+    if(onToggle(!isChecked)) {
```

Comment:
Why is needed? As much as I can think of, the component shouldn't aware about its consumer implementation. Once the user interacted with it, the UI should respond and then, call the callback. If, by any reason, the checkbox's state is not right, the consumer component will undo the change.

Reply:
This I did to show a confirm dialogue before changing the UI. Couldn't think of a better way to do this. I would be happy to know and implement the logic if you can suggest some pattern to tackle this kind of situation.